python/admin/api.py

```diff
@@ -0,0 +1,62 @@
+import os,subprocess
+from flask import Flask,request,abort,jsonify
+
+valid_properties = ["ssid", "channel", "hostname", "staticsite", "password", "system", "ui-config"]
+    
+
+def _call_command(extra_args):
+    cmd_args = ["sudo", "/usr/local/connectbox/bin/ConnectBoxManage.sh"]
+    called_cmd = subprocess.run(cmd_args + extra_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    result_string= called_cmd.stdout
+    if called_cmd.returncode != 0:
+        result_string= called_cmd.stderr
+
+    return jsonify(code=called_cmd.returncode,
+           result=result_string.decode("utf-8").rstrip().split("\n"))
+
+
+def get_property(prop):
+    prop_string = prop
+    if prop_string not in valid_properties:
+        abort(405)
+    return _call_command(["get", prop_string])
+
+
+def set_property(prop):
+    prop_string = prop
+    if prop_string not in valid_properties:
+        abort(400) # bad request
+    if (not request.json) or ("value" not in request.json):
+        abort(400) # bad request
+    return _call_command(["set", prop_string, request.json["value"]])
+
+
+def set_system_property():
+    if (not request.json) or ("value" not in request.json):
+        abort(400) # bad request
+
+    if request.json["value"] in ["shutdown", "reboot"]:
+        # Safe as input is limited
+        subprocess.Popen("sleep 1; " + request.json["value"], shell=True)

```

Comment:
what's the `sleep` doing here?

Reply:
Allowing the next line to run, completing the request. It does not block (i.e. forks)  so the next line (and going back up the stack to return the request) is run.